### PR TITLE
Update Octave to v4.2.1

### PR DESCRIPTION
**Pull request by obstschale**

### CODE REVIEW

**Comment by Kapeli** [comment]
Failed validation:

```
Checking Octave/versions/4.2.0/Octave.tgz...
File not found at "Commands-for-Completion.html" for entry with row id "269" of "Octave.docset"
File not found at "Commands-for-Completion.html" for entry with row id "270" of "Octave.docset"
File not found at "Commands-for-Completion.html" for entry with row id "271" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "449" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "450" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "451" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "728" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "729" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "730" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "731" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "732" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "733" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "1388" of "Octave.docset"
File not found at "Commands-for-Completion.html" for entry with row id "2062" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "2065" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "2331" of "Octave.docset"
File not found at "Commands-for-History.html" for entry with row id "2372" of "Octave.docset"
```

File: docsets/Octave/docset.json [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+      {
+        "version": "3.8.0",
+        "archive": "versions/3.8.0/Octave.tgz"
+      },
+      {
+        "version": "4.2.0",
+        "archive": "versions/4.2.0/Octave.tgz"
+      },
+      {
+        "version": "4.2.1",
```
Comment (Kapeli) [inline docsets/Octave/docset.json]: These need to be ordered by newest-first, not last.

Comment (obstschale) [inline docsets/Octave/docset.json]: :+1: Will fix this

**Comment by obstschale** [comment]
What does this validation fail mean?

**Comment by Kapeli** [comment]
Inside the `docSet.dsidx` of the `Octave/versions/4.2.0/Octave.tgz` docset you have entries that point to files that don't exist.

**Comment by obstschale** [comment]
Oh ok. I try to fix that. Thx

**Comment by obstschale** [comment]
Can I run the validation myself without opening a new PR?

**Comment by Kapeli** [comment]
Yep, run this from the root of the repo:

```
wget http://kapeli.com/feeds/zzz/docsetcontrib.tgz && tar -xzf docsetcontrib.tgz && ./docsetcontrib --verify
```

**Comment by obstschale** [comment]
You're awesome :+1: It seems to work.

**Pull request closed without merging**